### ISSUE TITLE
Memory corruption

### ISSUE THREAD

**Issue by Flakebi** [supertuxkart/stk-code#1746]
It looks like stk randomly overwrites memory and sometimes it is noticeable. E. g. it crashed nearly every time I exited an arena (#1409) and one time, the skidding sound was overwritten by the last_lap_fanfare. I expect that's a memory corruption too because I wasn't able to reproduce it the next time.

**Comment by deveee** [comment]
I recently fixed #1409. Are you able to reproduce other problems with memory?

**Comment by Flakebi** [comment]
Thanks for fixing #1409, it works for me too.
Recently I had the bug again, that another sound is played instead of the one that should be played (probably the same as #1830). This time the bowling ball sound was replaced by an engine...
When I finished the game (I stroke the opponent 3 times), stk crashed but I wasn't able to repeat it afterwards inside a debugger. :confused:

Though it's a nasty bug it happens really rarely (I got it two or three times in more than six months) so I don't think it's necessary to fix it before a release.

**Comment by Flakebi** [comment]
I finally got a stacktrace of this bug: https://gist.github.com/Flakebi/5539eebd0cfa4771bdcc

There was no output, only a segmentation fault.

Some more information:
```
In MusicOggStream::streamIntoBuffer:
 m_oggStream = {datasource = 0x0, seekable = 0, offset = 0, end = 0, oy = {data = 0x0, 
    storage = 0, fill = 0, returned = 0, unsynced = 0, headerbytes = 0, bodybytes = 0}, links = 0, 
  offsets = 0x0, dataoffsets = 0x0, serialnos = 0x0, pcmlengths = 0x0, vi = 0x0, vc = 0x0, 
  pcm_offset = 0, ready_state = 0, current_serialno = 0, current_link = 0, bittrack = 0, 
  samptrack = 0, os = {body_data = 0x0, body_storage = 0, body_fill = 0, body_returned = 0, 
    lacing_vals = 0x0, granule_vals = 0x0, lacing_storage = 0, lacing_fill = 0, lacing_packet = 0, 
    lacing_returned = 0, header = '\000' <repeats 281 times>, header_fill = 0, e_o_s = 0, 
    b_o_s = 0, serialno = 0, pageno = 0, packetno = 0, granulepos = 0}, vd = {analysisp = 0, 
    vi = 0x0, pcm = 0x0, pcmret = 0x0, pcm_storage = 0, pcm_current = 0, pcm_returned = 0, 
    preextrapolate = 0, eofflag = 0, lW = 0, W = 0, nW = 0, centerW = 0, granulepos = 0, 
    sequence = 0, glue_bits = 0, time_bits = 0, floor_bits = 0, res_bits = 0, 
    backend_state = 0x0}, vb = {pcm = 0x0, opb = {endbyte = 0, endbit = 0, buffer = 0x0, 
      ptr = 0x0, storage = 0}, lW = 0, W = 0, nW = 0, pcmend = 0, mode = 0, eofflag = 0, 
    granulepos = 0, sequence = 0, vd = 0x0, localstore = 0x0, localtop = 0, localalloc = 0, 
    totaluse = 0, reap = 0x0, glue_bits = 0, time_bits = 0, floor_bits = 0, res_bits = 0, 
    internal = 0x0}, callbacks = {read_func = 0x0, seek_func = 0x0, close_func = 0x0, 
    tell_func = 0x0}}

m_buffer_size = 44100

m_fileName = ""

m_playing = false


In MusicInformation::update:
m_title = "Penguin Party"
```

**Comment by hiker** [comment]
I am working on this atm. I have a suspicion what the reason is, but if you get this again, could you do a "thread apply all bt full" to get info from all threads.

**Comment by hiker** [comment]
Should be fixed in c37e1eec0df53433cf1cffdd0b263bbcd8169179 to 1e994748cc3f7c86ead7c9d58084bb20f97389ea (suspected reason: two threads accessing audio functions that are not thread safe at the same time). Please reopen with new complete (thread apply) stack trace if this should still happen
